File: tmp/security/authorization/src/main/java/org/tsdes/intro/spring/security/authorization/db/UserService.java
package org.tsdes.intro.spring.security.authorization.db;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.Collections;
import java.util.Set;

/**
 * Created by arcuri82 on 13-Dec-17.
 */
@Service
@Transactional
public class UserService {

    @Autowired
    private UserRepository userCrud;

    @Autowired
    private PasswordEncoder passwordEncoder;


    public boolean createUser(String username, String password) {

        try {
            String hashedPassword = passwordEncoder.encode(password);

            if (userCrud.exists(username)) {
                return false;
            }

            UserEntity user = new UserEntity();
            user.setUsername(username);
            user.setPassword(hashedPassword);
            user.setRoles(Collections.singleton("USER"));
            user.setEnabled(true);

            userCrud.save(user);

            return true;

        } catch (Exception e) {
            return false;
        }
    }
}

File: spring/security/distributed-session/zuul/src/main/kotlin/org/tsdes/spring/security/distributedsession/zuul/db/UserService.kt
package org.tsdes.spring.security.distributedsession.zuul

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.repository.CrudRepository
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

/**
 * Created by arcuri82 on 08-Nov-17.
 */
@Service
@Transactional
class UserService(
        @Autowired
        private val userCrud: UserRepository,
        @Autowired
        private val passwordEncoder: PasswordEncoder
){


    fun createUser(username: String, password: String, roles: Set<String> = setOf()) : Boolean{

        try {
            val hash = passwordEncoder.encode(password)

            if (userCrud.exists(username)) {
                return false
            }

            val user = UserEntity(username, hash, roles.map{"ROLE_$it"}.toSet())

            userCrud.save(user)

            return true
        } catch (e: Exception){
            return false
        }
    }

}

interface UserRepository : CrudRepository<UserEntity, String>

